monkey/monkey_island/cc/services/authentication_service/configure_flask_security.py

```diff
@@ -0,0 +1,121 @@
+import json
+import secrets
+from datetime import timedelta
+from pathlib import Path
+from typing import Any, Dict
+
+from flask.sessions import SecureCookieSessionInterface
+from flask_mongoengine import MongoEngine
+from flask_security import ConfirmRegisterForm, MongoEngineUserDatastore, Security, UserDatastore
+from wtforms import StringField, ValidationError
+
+from common.utils.file_utils import open_new_securely_permissioned_file
+from monkey_island.cc.mongo_consts import MONGO_DB_HOST, MONGO_DB_NAME, MONGO_DB_PORT, MONGO_URL
+
+from . import AccountRole
+from .role import Role
+from .user import User
+
+SECRET_FILE_NAME = ".flask_security_configuration.json"
+AUTH_EXPIRATION_TIME = 30 * 60  # 30 minutes authentication token expiration time
+
+
+def setup_authentication(app, data_dir):
+    app.config["MONGO_URI"] = MONGO_URL
+    app.config["MONGODB_SETTINGS"] = [
+        {
+            "db": MONGO_DB_NAME,
+            "host": MONGO_DB_HOST,
+            "port": MONGO_DB_PORT,
+        }
+    ]
```

Comment:
Extract mongoDB related setup to a separate method to make it easier to find and refactor once we switch database enginges?